logits_processor_zoo/vllm/trigger_phrase.py

```diff
@@ -40,14 +40,15 @@ def __init__(self, phrase: str, trigger_token_phrase: str, tokenizer: PreTrained
         self.initial_trigger_count = trigger_count
         self.trigger_after = trigger_after
         self.very_large_number = 999
-        # Initialize state for a new sequence
+        self._reset()
+
+    def _reset(self):
         self.index = -1
-        self.trigger_count = trigger_count
+        self.trigger_count = self.initial_trigger_count
 
     def __call__(self, prompt_tokens_ids: List[int], past_token_ids: List[int], scores: torch.Tensor) -> torch.Tensor:
-        if not past_token_ids:
-            self.index = -1
-            self.trigger_count = self.initial_trigger_count
+        if not past_token_ids:  # new generation
```

Comment:
Why aren't you using a similar base class for vllm?

Reply:
vllm provides past_token_ids. If it is an empty list or None, we can be sure that it is a new generation. In transformers, we actually make an hack to figure out new generations (new batches), which dont guarantee 100% correctness.